### ISSUE TITLE
Add more Description to thrift Program circular include error.

### ISSUE THREAD

**Issue by muandrew** [Microsoft/thrifty#266]
I'd like to change `loadIncludePrograms` from taking a `MutableSet<Program>` to taking a `MutableMap<Program, Program>` where the key stands for the a `Program` added to the map and the value stands for the parent that included the `Program`. Then if a cycle has been detected the error can print out the cycle.
```
    /**
     * Loads this program's symbol table and list of included Programs.
     * @param loader
     * @param visited
     */
    internal fun loadIncludedPrograms(loader: Loader, visited: MutableSet<Program>) {
        if (!visited.add(this)) {
            if (includedPrograms == null) {
                loader.errorReporter().error(location, "Circular include; file includes itself transitively")
                throw IllegalStateException("Circular include: " + location.path
                        + " includes itself transitively")
            }
            return
        }
```

**Comment by benjamin-bader** [comment]
Sounds like a nice feature!  Want to take a stab at it?

**Comment by benjamin-bader** [comment]
Fixed with #273; will be shipped in the next release.
